nibabel/cifti2/cifti2_axes.py

```diff
@@ -0,0 +1,1340 @@
+        if other_axis.unit != self.unit:
+            raise ValueError('Can only concatenate Series with the same unit')
+        return Series(self.start, self.step, self.size + other_axis.size, self.unit)
+
+    def __len__(self):
+        return self.size
+
+    def __eq__(self, other):
+        """
+        True if start, step, size, and unit are the same.
+        """
+        return (
+                isinstance(other, Series) and
+                self.start == other.start and
+                self.step == other.step and
+                self.size == other.size and
+                self.unit == other.unit
+        )
+
+    def __add__(self, other):
+        """
+        Concatenates two Series
+
+        Parameters
+        ----------
+        other : Series
+            Time Series to append at the end of the current time Series.
+            Note that the starting time of the other time Series is ignored.
+
+        Returns
+        -------
+        Series
+            New time Series with the concatenation of the two
+
+        Raises
+        ------
+        ValueError
+            raised if the repetition time of the two time Series is different
+        """
+        if isinstance(other, Series):
+            return self.extend(other)
+        return NotImplemented
+
+    def __getitem__(self, item):
+        if isinstance(item, slice):
+            step = 1 if item.step is None else item.step
+            idx_start = ((self.size - 1 if step < 0 else 0)
+                         if item.start is None else
+                         (item.start if item.start >= 0 else self.size + item.start))
+            idx_end = ((-1 if step < 0 else self.size)
+                       if item.stop is None else
+                       (item.stop if item.stop >= 0 else self.size + item.stop))
+            if idx_start > self.size:
+                idx_start = self.size - 1
+            if idx_end > self.size:
+                idx_end = self.size
+            nelements = (idx_end - idx_start) // step
+            if nelements < 0:
+                nelements = 0
+            return Series(idx_start * self.step + self.start, self.step * step,
+                          nelements, self.unit)
+        elif isinstance(item, integer_types):
+            return self.get_element(item)
+        raise IndexError('Series can only be indexed with integers or slices ' +
+                         'without breaking the regular structure')
+
+    def get_element(self, index):
+        """
+        Gives the time point of a specific row/column
+
+        Parameters
+        ----------
+        index : int
+            Indexes the row/column of interest
+
+        Returns
+        -------
+        float
+        """
+        if index < 0:
+            index = self.size + index
+        if index >= self.size:
+            raise IndexError("index %i is out of range for Series with size %i" %
+                             (index, self.size))
```

Comment:
If you're going to report the index, I'd suggest reporting the originally passed index, so that negative indices don't get transformed before being reported. Otherwise if you have a series with 9 elements and ask for -1, you'll get "index -2 is out of range for Series with size 9".

Reply:
Done